### PR TITLE
Stop scheduler halting when Retrofit experiences an issue

### PR DESCRIPTION
**Pull request by robfletcher**

### CODE REVIEW

File: keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceCheckScheduler.kt [hunk 1]
```diff
@@ -34,6 +36,7 @@ class ResourceCheckScheduler(
   @Scheduled(fixedDelayString = "\${keel.resource-check.frequency:PT1S}")
   fun checkResources() {
     if (enabled) {
+      publisher.publishEvent(ScheduledCheckStarting)
```
Comment (emjburns) [inline keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceCheckScheduler.kt]: Why the change to publishing an event here?

Comment (robfletcher) [reply keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceCheckScheduler.kt]: Needed something for the test to listen to to confirm the scheduler does not hang.

I figure tacking a health check to this probably makes sense too.

File: keel-api/src/test/kotlin/com/netflix/spinnaker/keel/integration/SchedulingResilienceTests.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  lateinit var service: DummyRetrofitService
+
+  @Test
+  fun `retrofit call completes even if an interceptor throws an exception`() {
+    val latch = CountDownLatch(1)
+    GlobalScope.launch {
+      try {
+        service.greet().await()
+      } finally {
+        latch.countDown()
```
Comment (robfletcher) [inline keel-api/src/test/kotlin/com/netflix/spinnaker/keel/integration/SchedulingResilienceTests.kt]: Don't care if the method returns or throws an exception. Just ensuring it doesn't hang.

File: keel-retrofit/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/KeelRetrofitConfiguration.kt [hunk 3]
```diff
@@ -82,18 +83,22 @@ class KeelRetrofitConfiguration {
...
             v.ifPresent {
               requestBuilder.addHeader(k, it)
             }
           }
       }
-      AuthenticatedRequest.allowAnonymous { chain.proceed(requestBuilder.build()) }
+      try {
+        AuthenticatedRequest.allowAnonymous { chain.proceed(requestBuilder.build()) }
+      } catch (ex: RuntimeException) {
+        throw IOException(ex)
```
Comment (robfletcher) [inline keel-retrofit/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/KeelRetrofitConfiguration.kt]: This is a workaround we could remove once kork#305 is merged
